src/wp-includes/class-wp-query.php

```diff
@@ -3636,6 +3644,7 @@ public function the_post() {
 		}
 
 		$this->in_the_loop = true;
+		$this->before_loop = false;
```

Comment:
We should also setup this property in `init` method.

Reply:
Done.